docs/enhancements/install_logs.md

```diff
@@ -0,0 +1,88 @@
+1. Hive users must get the OpenShift install log through Hive instead of going directly to the storage. When OpenShift install logs are written to S3, then a Hive administrator can go directly to S3 to get the logs.
+1. OpenShift install logs are relatively small, but minimum PV sizes are usually large. For instance, in AWS, the minimum size of a PV is 1 GiB, but OpenShift install logs tend to be less than 100 MiB.
+1. It is harder to maintain backups of the OpenShift install logs since individual PVs aren't easily backed up. By contrast, there are multiple programs that can maintain backups of S3 buckets. Also, S3 itself can be setup to replicate a bucket to other regions, has file change versioning and overall maintains a high level of resiliency.
+
+### Goals
+Store OpenShift install logs inside of S3 blob storage.
+
+### Non-Goals
+None
+
+## Open Questions
+1. What is the best way to pass the S3 parameters from the Hive operator to Hive controllers to the OpenShift install pod?
+   1. MUST allow for other blob storage machanisms to be added in the future.
+   1. We'll look to how other projects like Velero do this as a reference.
+1. What object should store the URL of the OpenShift install log in S3?
+
+## Proposal
+Change Hive to store OpenShift install logs in S3 blob storage instead of persistent volumes.
+
+Ensure that future enhancements can use similar mechanisms to upload to other blog storage systems.
+
+## User Stories
+
+### Story 1
+As a Hive administrator,
+I want OpenShift install logs to be stored in S3 compatible blob storage,
+So that simultaneous Hive triggered OpenShift installs can scale with available compute and memory instead of PVs.
+
+### Story 2
+As an Hive Developer,
+I want OpenShift install logs to be stored in S3 compatible blob storage,
+So that while developing, I can use Minio to capture OpenShift install logs.
+
+
+## Implementation Details / Notes / Constraints
+1. Configure S3 URL and credentials in HiveConfig. Have these passed down into OpenShift install pods. From the OpenShift install pods, upload the OpenShift install log to S3.
+  1. If no S3 information is given, do not gather OpenShift install logs.
+1. Remove all PV logic from the Hive code that launches the OpenShift install pod.
+1. Update the log retrieval script to gather the logs from S3 possibly using the URL and credentials in HiveConfig
+1. Update Hive documentation for troubleshooting failed OpenShift installs to point to the OpenShift install logs in S3.
+1. Store the URL of the OpenShift install log in S3 in a TBD Hive object.
+1. Prune the OpenShift install log when the Hive object is removed.
+
+## Risks and Mitigations
+There is a risk of using a large amount of data in S3 and having these OpenShift install logs live forever.
+
+Mitigations:
+- Clean up OpenShift install logs when Hive objects removed
```

Comment:
I agree. One scenario where it would be helpful to keep the install logs is for failures of the CI tests that run on stage and prod. Currently, if those installs fail, we have a short window of time to capture the logs before the test ends. It would be nice if those logs persisted for longer.

Reply:
/cc @jeefy 
we talked about persisting these logs in CICD.
what do you need to be able to access these?

Reply:
We have an AWS account that we use to push metrics to an S3 bucket. I think all we'd really need is the S3 bucket/file path and read access to that bucket.

Reply:
That sounds good.

How this works is that Hive is given creds and the url to the S3 bucket where to store them. We then put the URL of what we uploaded to our clusterprovision object.

So if someone has readonly creds to the bucket we were given and has access to the clusterprovision object, then they should be able to get the logs.